cowprotocol/cow-amm
# CoW AMM as a standalone contract

For gas efficiency, we want to make the CoW AMM a standalone contract and not rely on the ConditionalCoW framework outside of the watchtower for creating orders.

This is a PR that gives an overview of all changes needed for achieving that. It's not supposed to be reviewed but just used to give more legibility on what has been done so far for this task.
Eventually, it will be merged onto the main branch while preserving history.

### How to test

Verify full test coverage (if you see `approveUnlimited` not covered, update Foundry; support for constructor coverage has literally been implemented 3 days ago) and that all desired functionalities are there.
